pkg/workspace/templates.go

```diff
@@ -518,6 +518,11 @@ func ValidateProjectName(s string) error {
 		return errors.New("A project name must be 100 characters or less")
 	}
 
+	// This is needed to stop cyclic imports in DotNet projects
+	if strings.HasPrefix(strings.ToLower(s), "pulumi") {
```

Comment:
I feel like this condition is too prohibitive. I expect people to do things like `pulumi-test` and I would love to avoid breaking them. I would prohibit literally `Pulumi` and starting with `Pulumi.`, case sensitive.

Reply:
Fixed